text/0258-env-context-baggage-carriers.md

```diff
@@ -0,0 +1,236 @@
+
+## Internal details
+
+Defining the specification for Environment Variables as carriers for context
+will enable SDK's and other tools to implement getters and settings of context
+in a standard, observable way. Therefore, current OpenTelemetry language
+maintainers will need to develop language specific implementations that adhere
+to the specification. Of course the specification itself will have to be
+defined, and a decision made on whether or not to adhere to the W3C
+specification or not will need to be made.
+
+Due to the current programming conventions, prior art, and information below,
+it is recommended to leverage upper-cased environment variables for the
+carrier. `TextMapPropagator` will be the means of propagation for environment
+variables.
+
+### UNIX
+
+UNIX system utilities use upper-case for environment variables and lower-case
+are reserved for applications. Using upper-case will prevent conflicts with
+internal application variables.
+
+Environment variable names used by the utilities in the XCU specification
+consist solely of upper-case letters, digits and the "_" (underscore) from the
+characters defined in Portable Character Set. Other characters may be permitted
+by an implementation; applications must tolerate the presence of such names.
+Upper- and lower-case letters retain their unique identities and are not folded
+together. The name space of environment variable names containing lower-case
+letters is reserved for applications. Applications can define any environment
+variables with names from this name space without modifying the behaviour of
+the standard utilities.
+
+Source: [The Open Group, The Single UNIX® Specification, Version 2, Environment Variables](https://pubs.opengroup.org/onlinepubs/7908799/xbd/envvar.html)
+
+### Windows
+
+Windows is case-insensitive with environment variables. Despite this, the
+recommendation to be upper-cased across OS.
+
+Some languages already do this. This [CPython issue][cpython] discusses how
+Python automatically upper-cases environment variables. The issue was merged and
+this [documentation][cpython-doc] was added to clarify the behavior.
+
+[cpython]: https://github.com/python/cpython/issues/101754
+[cpython-doc]: https://docs.python.org/3/library/os.html#os.environ
+
+### Allowed characters
+
+The characters allowed in keys by the `TextMapPropagator` are also allowed as
+environment variable keys on UNIX and Windows.
+
+These are:
+
+> "!" / "#" / "$" / "%" / "&" / "'" / "*"
+> / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~"
+> / DIGIT / ALPHA
+
+To ensure compatibility, specification for Environment Variables SHOULD adhere
+to the current specification for `TextMapPropagator` where key/value pairs MUST
+only consist of US-ASCII characters that make up valid HTTP header fields as
+per RFC 7230.
+
+One key note is that windows disallows the use of the `=` character in
+environment variables. See [MS Env Vars][ms-env] for more information.
```

Comment:
> environment variables

names or values?

Reply:
only for names, so we are fine, value-wise (but this should be clarified).